server/data/data-jmap/src/main/java/org/apache/james/jmap/api/change/MailboxChange.java

```diff
@@ -0,0 +1,124 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.api.change;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.apache.james.jmap.api.vacation.AccountId;
+import org.apache.james.mailbox.model.MailboxId;
+
+public class MailboxChange {
+
+    public static class State {
```

Comment:
I guess it is ok for now, but when other changes will be implemented you might need to reuse the State and Limit classes... Might be better to extract them into their own class perhaps?

Reply:
We can still import static class anywhere, or am i misunderstanding your point here?

Reply:
Yes you can but I guess it would just be more clear, that's all :) but ok for now